offline_converter.py

```diff
@@ -224,9 +224,34 @@ def _convert(self, project: QgsProject) -> None:
                 project.removeMapLayer(layer)
                 continue
 
+            if layer_source.is_file and isascii(layer_source.filename):
+                self.warning.emit(
+                    self.tr("QFieldSync"),
+                    self.tr(
+                        'Layer "{}" stored at "{}" is not using ASCII encoded filename. Working with the layer might cause problems, please rename it using only ASCII charasters.'
+                    ).format(
+                        layer.name(),
+                        layer_source.filename,
+                    ),
+                )
+
             if layer_source.is_localized_path:
                 continue
 
+            if (
+                layer.type() == QgsMapLayer.VectorLayer
+                and layer.dataProvider().encoding() != "UTF-8"
+            ):
+                self.warning.emit(
+                    self.tr("QFieldSync"),
+                    self.tr(
+                        'Layer "{}" is not using UTF-8 encoding, but "{}". Working with the layer might cause problems, please convert it to UTF-8 layer.'
```

Comment:
I'm wondering whether we should limit this to known problematic datasets (i.e. as of now, I'd narrow this warning down to ESRI Shapefile datasets ATM)

Reply:
I would prefer people using utf8 all the time for any kind of data. It's 2021 :)

Reply:
Let's see. I suspect we'll get some feedback from users soon enough :)